addons/website_sale_coupon/models/sale_order.py

```diff
@@ -8,6 +8,26 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _try_pending_coupon(self):
+        if not request:
+            return
+
+        pending_coupon = request.session.get('pending_coupon', {})
+        status = {key: value for key, value in pending_coupon.items() if key == 'promo_code'}
```

Comment:
Isn't this just
```python
pending_coupon.pop('not_found')
pending_coupon.pop('error')
```

Reply:
This assumes no other keys will be added in `apply_coupon` which is not something I wanted to do in this method (even if it will still work fine even if keys are added).

Reply:
For example, the status can also hold a `generated_coupon` key